app/controllers/users_controller.rb

```diff
@@ -36,8 +36,12 @@ def edit
   #
   def update
     if current_user.update_attributes(update_user_params)
+      if params[:user][:regenerate_api_token]
+        current_user.regenerate_api_token
+      end
+
       respond_to do |format|
-        format.html { redirect_to root_path }
+        format.html { redirect_back(fallback_location: root_path) }
```

Comment:
When you click "regenerate token" you probably want to be taken back... I think this is fine, but I'm not sure where else the user settings page is?

Reply:
No where else than `/settings`

Reply:
Think it makes sense to just redirect to settings?